internal/stage9.go

```diff
@@ -18,19 +20,50 @@ func testpwd(stageHarness *test_case_harness.TestCaseHarness) error {
 		return err
 	}
 
-	command := "pwd"
 	cwd, err := os.Getwd()
 	if err != nil {
 		return fmt.Errorf("CodeCrafters internal error. Error getting cwd: %v", err)
 	}
 
 	testCase := test_cases.SingleLineOutputTestCase{
-		Command:                    command,
+		Command:                    "type pwd",
+		ExpectedPattern:            regexp.MustCompile(`^pwd is a( special)? shell builtin$`),
+		ExpectedPatternExplanation: fmt.Sprintf("match %q", (`pwd is a shell builtin`)),
+		SuccessMessage:             "Received current working directory response",
+	}
+	if err := testCase.Run(shell, logger); err != nil {
+		return err
+	}
+
+	var revertRenameOfPWD bool
+	path, err := exec.LookPath("pwd")
+	newPath := path + "Backup"
+	// On MacOS, the OS doesn't allow renaming the `pwd` binary
+	if err == nil && runtime.GOOS != "darwin" {
+		revertRenameOfPWD = true
```

Comment:
Would use `defer` for this to be safe (protect against any other `return err` statements added in the future). Also keeps the action and removal grouped together + you can remove `revertRenameOfPWD` entirely